src/stratigraphy/util/coordinate_extraction.py

```diff
@@ -0,0 +1,283 @@
+"""This module contains the CoordinateExtractor class."""
+
+import abc
+import logging
+from dataclasses import dataclass
+
+import fitz
+import regex
+
+logger = logging.getLogger(__name__)
+
+COORDINATE_ENTRY_REGEX = "(?:[12][\.\s']{0,2})?\d{3}[\.\s']{0,2}\d{3}\.?\d?"
+
+
+@dataclass
+class CoordinateEntry:
+    """Dataclass to represent a coordinate entry."""
+
+    first_entry: str | None = None
+    second_entry: str | None = None
+    coordinate_value: int | None = None
+
+    def __post_init__(self):
+        if self.coordinate_value is not None:
+            string_representation = str(self.coordinate_value)
+            if len(string_representation) < 6:  # leading zeros are lost in the int conversion
+                string_representation = "0" * (6 - len(string_representation)) + string_representation
+            self.first_entry = string_representation[:3]
+            self.second_entry = string_representation[3:]
+
+        elif self.first_entry is not None and self.second_entry is not None:
+            joint_coordinate = self.first_entry + self.second_entry
+            self.coordinate_value = int(joint_coordinate)
+        else:
+            raise ValueError("Either coordinate_value or first_entry and second_entry must be provided.")
+
+    def __repr__(self):
+        return f"{self.first_entry}'{self.second_entry}"
+
+
+@dataclass
+class Coordinate(metaclass=abc.ABCMeta):
+    """Abstract class for coordinates."""
+
+    east: CoordinateEntry
+    north: CoordinateEntry
+
+    def __post_init__(self):
+        # east always greater than north by definition. Irrespective of the leading 1 or 2
+        if self.east.coordinate_value < self.north.coordinate_value:
+            logger.info("Swapping coordinates.")
+            east = self.north
+            self.north = self.east
+            self.east = east
+
+    @abc.abstractmethod
+    def __repr__(self):  # noqa: D105
+        pass
+
+    @abc.abstractmethod
+    def to_json(self):
+        pass
+
+    @staticmethod
+    def from_json(input: dict):
+        east = input["E"]
+        north = input["N"]
+        if east > 2e6 and east < 1e7:
+            return LV95Coordinate(
+                CoordinateEntry(coordinate_value=east - 2e6), CoordinateEntry(coordinate_value=north - 1e6)
```

Comment:
Why subtract 2e6 / 1e6 here, instead of just leaving the 7-digit coordinate values?

Reply:
If we switch to purely int based coordinates I see 7 digits numbers favoured.